### PR TITLE
An atlernative for unified app settings endpoint.

### PR DESCRIPTION
**Pull request by jrmccannon**

### CODE REVIEW

File: src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs [hunk 1]
```diff
@@ -22,4 +22,5 @@ public interface IPasswordlessManagementClient
     Task<GetAppIdAvailabilityResponse> IsApplicationIdAvailableAsync(GetAppIdAvailabilityRequest request);
     Task EnableMagicLinksAsync(string appId, string performedBy);
     Task DisableMagicLinksAsync(string appId, string performedBy);
```
Comment (Tyrrrz) [inline src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs:24]: Would these be removed?

File: src/AdminConsole/Services/PasswordlessManagement/PasswordlessManagementClient.cs [hunk 2]
```diff
@@ -175,4 +175,9 @@ public async Task DisableMagicLinksAsync(string appId, string performedBy)
         response.EnsureSuccessStatusCode();
     }
 
+    public async Task SetAppSettingsAsync(string applicationId, SetAppSettingsRequest appSettings)
+    {
+        var response = await http.PostAsJsonAsync($"admin/apps/{applicationId}/settings", appSettings);
```
Comment (Tyrrrz) [inline src/AdminConsole/Services/PasswordlessManagement/PasswordlessManagementClient.cs]:
```suggestion
        using var response = await http.PostAsJsonAsync($"admin/apps/{applicationId}/settings", appSettings);
```

Comment (Tyrrrz) [inline src/AdminConsole/Services/PasswordlessManagement/PasswordlessManagementClient.cs]:
```suggestion
        var response = await http.PostAsJsonAsync($"admin/apps/{Uri.EncodeDataString(applicationId)}/settings", appSettings);
```

**Review by Tyrrrz — APPROVED** [review]

File: src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs (hunk 1 above)
Comment (jrmccannon) [reply src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs:24]: probably. unless we think its useful to keep them around

**Review by abergs — COMMENTED** [review]
I really like that the API becomes "PATCH"-like and accepts partial settings to change. 

I'm also in the "Keep things simple"-camp, preferring we keep it POST instead of going all in with HTTP verbs (since most developers don't know/care about the semantics), at least convinced otherwise.

There were some cleanup with the `switch`-statements, but other than that: Great work

File: src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs (hunk 1 above)
Comment (abergs) [reply src/AdminConsole/Services/PasswordlessManagement/IPasswordlessManagementClient.cs:24]: I vote remove

**Comment by jrmccannon** [comment]
I cleaned up a lot of stuff. I consolidated all the settings endpoints into the `app/features` SetFeatures endpoint. I refactored the SetFeaturesRequest to behave like a "patched" object. I added some documentation to hopefully make things a little more clear and refactored tests to reflect the changes.

**Review by Tyrrrz — APPROVED** [review]

**Pull request merged**